File: redsheet-portal/src/app/database-models/project-requirement.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectModule, TheirRequirementDetail, OurRequirementDetail, TheirBatna, Ourbatna,  } from './'
export class ProjectRequirement {
    constructor(projectRequirement?: ProjectRequirement )  {
        let properties = [ "isZoma", "ourStrategy", "projectRequirementId", "projectModuleId", "requirementCategoryId", "theirRequirementDetails", "ourRequirementDetails", "theirBatnas", "ourbatnas",];
        for (let property of properties)
            if (projectRequirement && projectRequirement[property])
                this[property] = projectRequirement[property];
    }
 
	isZoma : boolean = false ;
 
	@required()
    @maxLength(500)
	ourStrategy : string =   undefined;
 
	projectRequirementId : number =   0 ;
 
    @range(0,2147483647)
	projectModuleId : number =   undefined;
	projectModule : ProjectModule  ;
 
    @range(0,2147483647)
	requirementCategoryId : number =   undefined;
	@nested(TheirRequirementDetail)
	theirRequirementDetails: TheirRequirementDetail[];

	@nested(OurRequirementDetail)
	ourRequirementDetails: OurRequirementDetail[];

	@nested(TheirBatna)
	theirBatnas: TheirBatna[];

	@nested(Ourbatna)
	ourbatnas: Ourbatna[];



}

File: RedSheet.DbEntities/Models/ProjectRequirement.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("ProjectRequirements", Schema = "dbo")]
	[Serializable]
	public partial class ProjectRequirement  
	{
 
	[Required]
	public bool IsZoma { get; set; } 
 
	[MaxLength(500)]
	public string OurStrategy { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ProjectRequirementId { get; set; } 
 
	[RelationshipTableAttribue("ProjectModules","dbo")]
	//Relationships
	public int ProjectModuleId { get; set; } 
 
	[ForeignKey("ProjectModuleId")]
	public ProjectModule ProjectModule { get; set; } 
 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public RequirementCategory RequirementCategoryId { get; set; } 
	[InverseProperty("ProjectRequirement")]
	public ICollection<TheirRequirementDetail> TheirRequirementDetails { get; set; } 

	[InverseProperty("ProjectRequirement")]
	public ICollection<OurRequirementDetail> OurRequirementDetails { get; set; } 

	[InverseProperty("ProjectRequirement")]
	public ICollection<TheirBatna> TheirBatnas { get; set; } 

	[InverseProperty("ProjectRequirement")]
	public ICollection<Ourbatna> Ourbatnas { get; set; } 


	public ProjectRequirement()
	{
	this.TheirRequirementDetails = new HashSet<TheirRequirementDetail>();

	this.OurRequirementDetails = new HashSet<OurRequirementDetail>();

	this.TheirBatnas = new HashSet<TheirBatna>();

	this.Ourbatnas = new HashSet<Ourbatna>();

    }

    }
}

